apex-squid/src/main/java/org/fundacionjala/enforce/sonarqube/apex/api/ApexGrammarBuilder.java

```diff
@@ -0,0 +1,292 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.fundacionjala.enforce.sonarqube.apex.api;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+import com.sonar.sslr.api.Grammar;
+import com.sonar.sslr.api.TokenType;
+import com.sonar.sslr.api.typed.GrammarBuilder;
+
+import org.sonar.sslr.grammar.GrammarRuleKey;
+import org.sonar.sslr.grammar.LexerfulGrammarBuilder;
+import org.sonar.sslr.grammar.LexerlessGrammarBuilder;
+import org.sonar.sslr.internal.vm.FirstOfExpression;
+import org.sonar.sslr.internal.vm.OptionalExpression;
+import org.sonar.sslr.internal.vm.ParsingExpression;
+import org.sonar.sslr.internal.vm.PatternExpression;
+import org.sonar.sslr.internal.vm.SequenceExpression;
+import org.sonar.sslr.internal.vm.StringExpression;
+import org.sonar.sslr.internal.vm.lexerful.TokenTypeExpression;
+import org.sonar.sslr.internal.vm.lexerful.TokenValueExpression;
+
+/**
+ * Utility class for create a {@link GrammarBuilder} implementation.
+ */
+public class ApexGrammarBuilder {
+
+    /**
+     * Stores a error message when failing parser.
+     */
+    private static final String PARSING_ERROR_MESSAGE = "Incorrect type of parsing expression: %s";
+
+    /**
+     * Stores a identifier pattern.
+     */
+    private static final String IDENTIFIER_PATTERN = "[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?+";
+
+    /**
+     * Stores a error message when rule is null.
+     */
+    private static final String RULE_ERROR_MESSAGE = "Rules can't be null";
+
+    /**
+     * Represents a map of the rules and expression.
+     */
+    private final Map<GrammarRuleKey, ParsingExpression> mapRules;
+
+    /**
+     * Defines the current grammar type which is being used.
+     */
+    private final boolean isFulGrammar;
+
+    /**
+     * Represents a current rule.
+     */
+    private GrammarRuleKey currentRule;
+
+    /**
+     * Represents a root rule.
+     */
+    private GrammarRuleKey rootRule;
+
+    /**
+     * Create a apex grammar builder.
+     *
+     * @param isFulGrammar represents the type of grammar builder required.
+     * @return a ApexGramamrBuilder.
+     */
+    public static ApexGrammarBuilder create(boolean isFulGrammar) {
+        return new ApexGrammarBuilder(isFulGrammar);
+    }
+
+    /**
+     * Default constructor that initialize variables.
```

Comment:
Default constructor that initialize variables

Default constructor that initializes variables